timelock-corruption-detection/src/main/java/com/palantir/timelock/corruption/detection/HistoryAnalyzer.java

```diff
@@ -36,15 +43,49 @@ private HistoryAnalyzer() {
         // do not create instance of this class
     }
 
+    public static CorruptionHealthReport corruptionHealthReportForHistory(
+            List<CompletePaxosHistoryForNamespaceAndUseCase> history) {
+
+        Map<NamespaceAndUseCase, CorruptionCheckViolation> namespaceAndUseCaseCorruptionCheckViolationMap =
+                history.stream()
+                        .collect(Collectors.toMap(
+                                HistoryAnalyzer::extractNamespaceAndUseCase,
+                                HistoryAnalyzer::violatedCorruptionChecksForNamespaceAndUseCase));
+
+        SetMultimap<CorruptionCheckViolation, NamespaceAndUseCase> entryEntrySetMultimap = KeyedStream.stream(
```

Comment:
Let's name this based on what the map actually does: `namespacesExhibitingViolations` or something like this.